ddd-core/src/main/kotlin/io/paju/ddd/State.kt

```diff
@@ -12,4 +12,16 @@ inline fun <reified T: State>expectState(instance: State): T {
     }else{
         throw DddRuntimeException("Invalid instance type")
     }
+}
+
+fun AggregateRoot<*, *>.expectInitializedState() {
+    if(!isInitialized()) {
+        throw DddRuntimeException("Invalid instance type")
+    }
+}
+
+fun AggregateRoot<*, *>.expectUninitializedState() {
+    if(isInitialized()) {
```

Comment:
Should the exception message updated respectively? I.e "Instance not in expected state. Instance state was xxx, expected YYY"?

Reply:
Good catch! Fixed in next commit